src/features/wav_iterator.py

```diff
@@ -61,12 +62,23 @@ def batcher(feature_iter, batch_size=256):
         # grouped together
         try:
             batch_transposed = []
-            for dataset in list(zip(*new_batch)):
+            data_slice = list(zip(*new_batch))
+
+            # Code if we require the key to be returned
+            if return_key:
+                if type(feature_iter)!=mixer.FeatureMixer: 
```

Comment:
The mixer has an added layer of tuples, which is why this needs to be separated once more. If that's a simple transpose, then that's fine. However, that is why there is this conditional...I'd be open to understanding how your edits will do so in a general way...